### PR TITLE
convert signup tests from Protractor to Playwright

### PR DESCRIPTION
**Pull request by JeanneSon**
## Description
The following tests have been converted to playwright

Bellows E2E Signup app
✓ can verify required information filled
✓ cannot submit if email is invalid
✓ cannot submit if password is weak
✓ can submit if password not weak
✓ can submit if password is showing and not weak
✓ cannot submit if captcha not selected
✓ can submit a user registration request and captcha is invalid
✓ finds the admin user (case insensitive) already exists
✓ can prefill email address that can't be changed
✓ can prefill email address that already exists
✓ can signup a new user
✓ redirects to projects page if already logged in

For details/remarks regarding converting the above tests, refer to the spreadsheet; https://docs.google.com/spreadsheets/d/1w9wjfm-02QHLhv0ObPAKKlqnDWeag379GIXwOcRU4W0/edit#gid=0

Fixes # (no issue associated)

### Type of Change
E2E tests and signup page email-taken error text improvement

## Checklist

- [x] I have performed a self-review of my own code
- [x] I have reviewed the title/description of this PR which will be used as the squashed PR commit message
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

**Comment by JeanneSon** [comment]
The first test (Can verify required information filled) is kind of redundant with respect to the fourth test (Can submit if password not weak).

File: test/e2e/pages/signup.page.ts [hunk 1]
```diff
@@ -1,38 +1,97 @@
...
+        switch (await this.captcha.expectedItemName.innerText()) {
+          case 'Blue Square':
+            await this.captcha.blueSquareButton.click();
+            break;
+          case 'Yellow Circle':
+            await this.captcha.yellowCircleButton.click();
+            break;
+          case 'Red Triangle':
+            await this.captcha.redTriangleButton.click();
+        }
```
Comment (megahirt) [inline test/e2e/pages/signup.page.ts:68]: I find it mildly amusing that the point of a captcha is to make it difficult (impossible?) for a machine to programmatically exercise a form...and yet this block of code is doing exactly that.  I guess our captcha implementation is pretty old-fashioned given that it is this easy to program around 😆

File: test/e2e/signup.spec.ts [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+    await signupPage.passwordInput.fill(constants.passwordValid);
+    await signupPage.captcha.setInvalidCaptcha();
+    await expect(signupPage.signupButton).toBeEnabled();
+    await signupPage.passwordInput.fill(constants.passwordTooShort);
+    await signupPage.captcha.setInvalidCaptcha();
+    await expect(signupPage.passwordIsWeak).toBeVisible();
+    await expect(signupPage.signupButton).not.toBeEnabled();
+  });
+
+  test('Can submit if password not weak', async () => {
```
Comment (megahirt) [inline test/e2e/signup.spec.ts:50]:
```suggestion
  test('Can submit if the password is strong', async () => {
```

File: test/e2e/signup.spec.ts [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+test.describe('E2E Signup app', () => {
+  const constants = require('./testConstants.json');
+  let signupPage: SignupPage;
+
+  test.beforeAll(async ({ anonTab }) => {
+    signupPage = new SignupPage(anonTab);
+  })
+
+  // this name is not very descriptive
+  test('Can verify required information filled', async () => {
```
Comment (megahirt) [inline test/e2e/signup.spec.ts:14]:
```suggestion
  test('Can fill out the fields in the form', async () => {
```

File: test/e2e/signup.spec.ts [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+
+    await signupPage.nameInput.fill(constants.unusedName);
+    await signupPage.emailInput.fill(constants.unusedEmail);
+    await signupPage.passwordInput.fill(constants.passwordValid);
+    await expect(signupPage.signupButton).toBeDisabled();
+  });
+
+  // name is not good
+  // suggested name: invalid captcha allows submit but then displays error
+  test('Can submit a user registration request and captcha is invalid', async () => {
```
Comment (megahirt) [inline test/e2e/signup.spec.ts:84]: @JeanneSon you have my permission to improve the name of these tests.  When changing the names of tests from the protractor versions, please make a note in the PR description.

```suggestion
  test('Captcha is invalid but user can still submit form', async () => {
```

File: test/e2e/pages/signup.page.ts [hunk 5]
```diff
@@ -1,38 +1,97 @@
...
+    }
+    await expect(this.emailInput).toBeVisible();
   }
 
-  async goto() {
+  async gotoExpectRedirect(redirectedToUrl: string) {
     await this.page.goto(SignupPage.url);
-    await expect(this.emailInput).toBeVisible();
+    expect(this.page.url()).toContain(redirectedToUrl);
+  }
```
Comment (megahirt) [inline test/e2e/pages/signup.page.ts:89]: This function is only used once.  I am ambivalent about having a dedicated function for something that is likely only used in a single test (below).  This expect() clause could just as easily appear directly in the test.

**Review by megahirt — APPROVED** [review]
@JeanneSon great work and I have a few non-blocking suggestions.  Thanks!

File: test/e2e/signup.spec.ts [hunk 6]
```diff
@@ -0,0 +1,142 @@
...
+    await signupPage.goto();
+
+    await expect(signupPage.signupButton).toBeDisabled();
+    await signupPage.nameInput.fill(constants.unusedName);
+    await signupPage.emailInput.fill(constants.unusedEmail);
+    await signupPage.passwordInput.fill(constants.passwordValid);
+    await signupPage.captcha.setInvalidCaptcha();
+    await expect(signupPage.signupButton).toBeEnabled();
+  });
+
```
Comment (megahirt) [inline test/e2e/signup.spec.ts:24]: Per your suggestion, @JeanneSon , let's go ahead and delete this redundant test
```suggestion
```

**Pull request merged**